coins/bitcoin/src/bitcoin_wallet.rs

```diff
@@ -422,10 +469,15 @@ impl BitcoinWallet {
             .address_index(max_address + 1)
             .build()
             .to_string();
-        let next_hd_key = self.master_hd_key()?.derive(next_deriv_path)?;
+        let next_hd_key = self.master_hd_key()?.derive(&next_deriv_path)?;
         BitcoinAddress::from_hd_key(&next_hd_key, self.address_format)
     }
 
+    /// Sets the [master hd key][HDKey] associated with the [wallet][BitcoinWallet].
```

Comment:
What's the need for this setter?

Reply:
adding this function allows the master key to be set directly without going through the builder, without this function, only way to set this field is by using  the builder pattern

Reply:
ok no problem, Ill want to review this later